python/ray/scripts/scripts.py

```diff
@@ -2223,6 +2224,68 @@ def global_gc(address):
     print("Triggered gc.collect() on all workers.")
 
 
+@cli.command(hidden=True)
+@click.option(
+    "--address", required=False, type=str, help="Override the address to connect to."
+)
+@click.option(
+    "--node-id",
+    required=True,
+    type=str,
+    help="Hex id of the worker node to be drained.",
+)
+@click.option(
+    "--reason",
+    required=True,
+    type=click.Choice(
+        # Exclude 0 which is UNSPECIFIED
+        [item[0] for item in autoscaler_pb2.DrainNodeReason.items() if item[1] != 0]
+    ),
+    help="The reason why the node will be drained.",
+)
+@click.option(
+    "--reason-message",
+    required=True,
+    type=str,
+    help="The detailed drain reason message.",
+)
+@click.option(
+    "--deadline-timestamp-ms",
+    required=False,
+    type=int,
+    default=0,
+    help="Timestamp in ms when the node to be drained will be force killed. "
+    "Default is 0 which means there is no deadline",
+)
```

Comment:
Is this a timestamp or a time from right now? Also, `ms` is a weird choice for units here

Reply:
Timestamp.

> Also, ms is a weird choice for units here

@edoakes do you have any suggestions? In reality, seconds is probably enough, I just gave a bit more precisions just in case.

Reply:
Oh, so it's a timestamp since epoch? That is a really strange API / hard to use. How about making it seconds (as a float) from now and generate the timestamp in the CLI code?

Also, `-1` is more standard for "no deadline"

Reply:
`0` was chosen since it's the default value if you don't set it in the proto so 0 means no deadline is set.